code/modules/admin/verbs/plane_debugger.dm

```diff
@@ -99,234 +103,66 @@
-			var/relay_ref = "[string_plane]-[string_target]"
-			this_relay["our_ref"] = relay_ref
-			relay_deets[relay_ref] = this_relay
-			this_plane["outgoing_relays"] += relay_ref
-
-			// If we've already encoded our target plane, update its incoming relays list
-			// Otherwise, we'll handle this later
-			var/list/existing_target = plane_info[string_target]
-			if(existing_target)
-				existing_target["incoming_relays"] += relay_ref
-			else
-				var/list/pending_plane = pending_relays[string_target]
-				if(!pending_plane)
-					pending_plane = list()
-					pending_relays[string_target] = pending_plane
-				pending_plane += relay_ref
-
-		this_plane["incoming_filters"] = list()
-		this_plane["outgoing_filters"] = list()
-		// We're gonna collect a list of filters, partly because they're useful info
-		// But also because they can be used as connections, and we need to support that
-		for(var/filter_id in plane.filter_data)
+		for (var/filter_id in plane.filter_data)
 			var/list/filter = plane.filter_data[filter_id]
 			if(!filter["render_source"])
 				continue
+
 			var/list/filter_info = filter.Copy()
-			filter_info["target_ref"] = string_plane
 			filter_info["name"] = filter_id
-			filter_queue += list(filter_info)
-
-		plane_info[plane_string] = this_plane
-		render_target_to_plane[plane.render_target] = this_plane
-
-	for(var/list/filter in filter_queue)
-		var/source = filter["render_source"]
-		var/list/source_plane = render_target_to_plane[source]
-		var/list/target_plane = plane_info[filter["target_ref"]]
-		var/source_ref = source_plane["our_ref"]
-		filter["source_ref"] = source_ref
-		var/our_ref = "[source_ref]-[filter["target_ref"]]-filter"
-		filter["our_ref"] = our_ref
-		filter_connections[our_ref] = filter
-		source_plane["outgoing_filters"] += our_ref
-		target_plane["incoming_filters"] += our_ref
-
-	// Only load this once. Prevents leaving off orphaned components
-	if(!depth_stack[current_group])
-		depth_stack[current_group] = treeify(plane_info, relay_deets, filter_connections)
-
-	// We will use this js side to arrange our plane masters and such
-	// It's essentially a stack of where they should be displayed
-	data["depth_stack"] = depth_stack[current_group]
-	return data
+			filter_info["our_ref"] = "[plane.plane]-[filter_id]"
+			filters += list(filter_info)
 
-// Reading this in the queue tells the search to increase the depth, and then push another increase command to the end of the stack
-// This way we ensure groupings always stay together, and depth is respected
-#define COMMAND_DEPTH_INCREASE "increase_depth"
-#define COMMAND_NEXT_PARENT "next_parent"
-
-/// Takes a list of js formatted planes, and turns it into a tree based off the back connections of relays
-/// So start at the top master plane, and work down
-/// Haha jerry what if I added commands to my list parser lmao lol
-/datum/plane_master_debug/proc/treeify(list/plane_info, list/relay_info, list/filter_connections)
```

Comment:
rip some of the worst code I have ever written (human plane updating is a close contender tho)

Reply:
I still hate you for that code (I shall obliterate it with BSA when I get to it)